### PR TITLE
[RN] Drop the react-native-permissions dependency

### PR DESCRIPTION
**Pull request by saghul**
Depends on https://github.com/jitsi/react-native-webrtc/pull/39

### CODE REVIEW

File: android/sdk/src/main/java/org/jitsi/meet/sdk/ReactActivityLifecycleCallbacks.java [hunk 1]
```diff
@@ -126,4 +143,25 @@ public static void onNewIntent(Intent intent) {
             reactInstanceManager.onNewIntent(intent);
         }
     }
+
+    public static void onRequestPermissionsResult(
```
Comment (paweldomas) [inline android/sdk/src/main/java/org/jitsi/meet/sdk/ReactActivityLifecycleCallbacks.java]: All those arguments need to be documented, that includes the 'requestPermissions' method below

File: android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetActivity.java [hunk 2]
```diff
@@ -283,6 +291,14 @@ protected void onUserLeaveHint() {
         }
     }
 
+    /**
+     * Implementation of the {@code PermissionAwareActivity} interface.
+     */
+    @Override
+    public void requestPermissions(String[] permissions, int requestCode, PermissionListener listener) {
```
Comment (paweldomas) [inline android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetActivity.java]: where is this method called from ?

Comment (paweldomas) [reply android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetActivity.java]: oh now I see that PermissionAwareActivity comes from react

File: android/sdk/src/main/java/org/jitsi/meet/sdk/ReactActivityLifecycleCallbacks.java (hunk 1 above)
Comment (paweldomas) [reply android/sdk/src/main/java/org/jitsi/meet/sdk/ReactActivityLifecycleCallbacks.java]: Or maybe some link to Android documentations about permissions APIs used here. Those arguments kind of come from the Android's Activity#requestPermissions, right ?

Comment (saghul) [reply android/sdk/src/main/java/org/jitsi/meet/sdk/ReactActivityLifecycleCallbacks.java]: I'll add some docs. `onRequestPermissionsResult` comes from `Activity` and `requestPermissions` is an extension of Activity's method.

File: android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetActivity.java (hunk 2 above)
Comment (saghul) [reply android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetActivity.java]: Yep, it's part of the interface.

**Comment by saghul** [comment]
@paweldomas Updated, PTAL.

**Comment by paweldomas** [comment]
@saghul could please update the react-native-webrtc version, so that we can see the tests ?

**Comment by saghul** [comment]
@paweldomas Done!

**Comment by saghul** [comment]
Jenkins please test this please.

**Pull request merged**